test/luatest_helpers/vtest.lua

```diff
@@ -535,10 +543,39 @@ local function router_new(g, name, cfg)
     g[name] = server
     g.cluster:add_server(server)
     server:start()
+    return server
+end
+
+--
+-- Create a new router in the cluster.
+--
+local function router_new(g, name, cfg)
+    local server = server_build_and_start(g, name)
     router_cfg(server, cfg)
     return server
 end
 
+--
+-- Create a new router inside fiber. pre_func is used to
+-- declare function, which will be run before starting
+-- router configuration. opts are options passed to pre_func.
+-- They are supposed to have the table type
+--
+local function router_fiber_new(g, name, cfg, pre_func, opts)
+    local server = server_build_and_start(g, name)
+    if pre_func then
+        server:exec(pre_func, opts)
+    end
+    router_fiber_cfg(server, cfg)
+    return server
+end
+
+local function router_fiber_new_join(router)
+    router:exec(function()
+        if f then f:join() end
```

Comment:
Does this mean you have a global variable called `f` on the instance? Lets not do that, especially in the generic vtest. You can declare such small common variables per-test-case. To avoid luacheck warnings, you can do it like that:
```Lua
server:exec(function()
    rawset(_G, 'myvar', 100)
end)
server:exec(function()
    assert(_G.myvar == 100)
end)
```